File: src/python/maps/map19.py
import ika

def AutoExec(engineRef):
    engineRef.background = engineRef.getImage('gfx/mountains.png')

def to16(engineRef):
    offset_from = 44 * 16  # first vertical pos possible
    offset_to = 4 * 16  # first vertical pos possible
    y = engineRef.player.sprite.y - offset_from + offset_to
    yield from engineRef.mapSwitchTask('map16.ika-map', (1 * 16, y))

def to21(engineRef):
    yield from engineRef.mapSwitchTask('map21.ika-map', (engineRef.player.sprite.x, 38 * 16))

File: src/winter/maps/map19.ts
import { MapScript } from "./mapscript.js"
import { PyEngine } from "./../winter.js"

export default new MapScript(autoexec, { to16, to21 })

export function autoexec(engineRef: PyEngine) {
    const engine = engineRef.getEngine()
    engineRef.pySetBackground(engine.getImage('gfx/mountains.png'))
}

export function *to16(engineRef: PyEngine) {
    const offset_from = 44 * 16  // first vertical pos possible
    const offset_to = 4 * 16  // first vertical pos possible
    const p = engineRef.getPlayerEntity()
    const y = p.sprite.y - offset_from + offset_to
    yield* engineRef.mapSwitchTask('map16.ika-map', [1 * 16, y])
}

export function *to21(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map21.ika-map', [p.sprite.x, 38 * 16])
}
